nameko/testing/services.py

```diff
@@ -76,33 +75,12 @@ def catch_container_errors(gt):
     yield hook
 
 
-class EntrypointWaiter(DependencyProvider):
-    """Helper for `entrypoint_waiter`
-
-    DependencyProvider to be manually (and temporarily) added to an existing
-    container. Takes an entrypoint name, and exposes a `wait` method, which
-    will return once the entrypoint has fired.
-    """
-
-    class Timeout(Exception):
-        pass
-
-    def __init__(self, entrypoint):
-        self.attr_name = '_entrypoint_waiter_{}'.format(entrypoint)
-        self.entrypoint = entrypoint
-        self.done = Semaphore(value=0)
-
-    def worker_teardown(self, worker_ctx):
-        entrypoint = worker_ctx.entrypoint
-        if entrypoint.method_name == self.entrypoint:
-            self.done.release()
-
-    def wait(self):
-        self.done.acquire()
+class EntrypointWaiterTimeout(Exception):
+    pass
 
 
 @contextmanager
-def entrypoint_waiter(container, entrypoint, timeout=30):
+def entrypoint_waiter(container, method_name, timeout=30, callback=None):
```

Comment:
i'm not a huge fan of this `callback` api. do you have other use-cases in mind than multiple calls?